### PR TITLE
Drop kdm and kde4 modules

### PR DESCRIPTION
**Pull request by grahamc**
###### Motivation for this change

re: https://github.com/NixOS/nixpkgs/issues/15866

cc: @ttuegel @vcunat @globin @domenkozar @dezgeg @fpletz @edolstra 

One thing I'm not sure about is reverting @Phreedom's pinning of Quassel to KDE4.

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by ttuegel** [comment]
> One thing I'm not sure about is reverting @Phreedom's pinning of Quassel to KDE4.

That should be fine.

**Comment by grahamc** [comment]
I'd love it if I got a 👍 from @globin and/or @edolstra before merging this, but I'm getting a bit itchy to do it anyway :)

**Comment by peterhoeg** [comment]
As a data point on quassel, I'm using the qt5 version and it seems fine. In all fairness my total experience with IRC clients in general is a whopping 12 hours so I may not have exercised all the features...

**Comment by globin** [comment]
:+1:

File: nixos/modules/services/networking/quassel.nix [hunk 1]
```diff
@@ -25,12 +25,12 @@ in
 
       package = mkOption {
         type = types.package;
-        default = pkgs.kde4.quasselDaemon;
-        defaultText = "pkgs.kde4.quasselDaemon";
+        default = pkgs.quasselDaemon_qt5;
```
Comment (kierdavis) [inline nixos/modules/services/networking/quassel.nix]: I might be missing something here but it doesn't looks like `quasselDaemon_qt5` exists in `all-packages.nix`. I think this should `quasselDaemon` (which is still qt5).

Comment (kierdavis) [inline nixos/modules/services/networking/quassel.nix]: See #22853.

**Pull request merged**